pkg/rbac/clusterScopedAuthorization.go

```diff
@@ -7,14 +7,17 @@ import (
 
 	"github.com/doug-martin/goqu/v9"
 	"github.com/stolostron/search-v2-api/pkg/config"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
 	"k8s.io/klog/v2"
 )
 
 type clusterScopedResources struct {
-	err       error
-	resources map[string][]string
-	updatedAt time.Time
-	lock      sync.Mutex
+	err        error
+	resources  map[string][]string
+	namespaces []string
+	updatedAt  time.Time
+	lock       sync.Mutex
 }
```

Comment:
- I suggest that we rename this object (and file) now that it has a wider scope.
- I also did some renaming and organization, we'll need to update the rest of this file accordingly.
```go
// Cache data shared across all users.
type SharedData struct {
	// These are the data fields.
	csResources  map[string][]resource  // Cluster-scoped resources (ie. Node)
	namespaces   []string
	
	// These are internal objects to track the state of the cache.
	csErr         error        // Capture errors retrieving cluster-scoped resources.
	csLock        sync.Mutex   // Locks the csResources map while updating it.
	csUpdatedAt   time.Time    // Time when cluster-scoped data was last updated.
	nsErr         error        // Capture errors retrieving namespaces.
	nsLock        sync.Mutex   // Locks the namespaces array while updating it.
	nsUpdatedAt   time.Time    // Time when namespaces data was last updated.
}
```

This `resource` struct will be used here and also for namespaced resources, so maybe define it in a common file.
```go
// A Kubernetes resource is identified by its apigroup and kind.
// Resource version is not considered for RBAC.
type resource struct {
	apigroup.  string
	kind       string
}
```